tests/test_dataset.py

```diff
@@ -0,0 +1,122 @@
+# ruff: noqa: S105, S106, SLF001
+
+from unittest.mock import MagicMock, patch
+
+import pyarrow as pa
+import pytest
+from pyarrow import fs
+
+from timdex_dataset_api.dataset import DatasetNotLoadedError, TIMDEXDataset
+
+
+@pytest.mark.parametrize(
+    ("location", "expected_filesystem", "expected_source"),
```

Comment:
What is the difference between the "location" and "expected source" parametrized arguments? Is the main difference the removal of the "s3://" prefix for parquet files on S3?

Reply:
Yep. Pyarrow uses "source" when talking about loading a dataset, which is why `parse_location()` returns the tuple `(filesystem, source)`.  It's just a normalized form of the location.